test/lib/helpers/doChannelReportTest.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const assert          = require('assert');
+const sinon           = require('sinon');
+const mockBot         = require('../../support/mockBot');
+const dbHelper        = require('../../support/persistenceHelper');
```

Comment:
My bad on language and naming. This is an integration test, not a unit test. This is different from an acceptance or ui test which is what cucumber should be used for. Martin Fowler makes this distinction differently in his version of the testing pyramid: https://martinfowler.com/bliki/TestPyramid.html. Regardless of naming there should be a small number of tests that drive the application from the outside; a medium number of tests that test sections of code to ensure they are working together correctly as a system. Everything outside that system gets mocked. Then there should be lots of unit tests which focus on a single object or function and mock everything outside of that universe.

So, an integration test was absolutely essential to me understanding this code and ensuring that it worked the same while I refactored and built out unit tests.